openedx/core/djangoapps/user_authn/views/utils.py

```diff
@@ -108,3 +111,60 @@ def get_mfe_context(request, redirect_to, tpa_hint=None):
         'countryCode': country_code,
     })
     return context
+
+
+def generate_username(username_initials):
+    """
+    Generate a username based on initials and current date.
+
+    Args:
+    - username_initials (str): Initials used for constructing the username.
+
+    Returns:
+    - str: Generated username.
+    """
+    current_year = datetime.now().year % 100
+    current_month = datetime.now().month
+
+    random_string = ''.join(random.choices(
+                            string.ascii_letters +
+                            string.digits,
+                            k=settings.RANDOM_USERNAME_STRING_LENGTH))
+
+    username = f"{username_initials}_{current_year:02d}{current_month:02d}_{random_string}"
+    return username
+
+
+def generate_username_from_data(data):
+    """
+    Generate a username based on the provided data.
+
+    Args:
+    - data (dict): Dictionary containing user data, including 'first_name', 'last_name', or 'name'.
+
+    Returns:
+    - str: Generated username.
+    """
+    try:
+        if 'first_name' in data and 'last_name' in data:
+            first_name = data['first_name']
+            last_name = data['last_name']
+
+            username_initials = f"{first_name[0]}{last_name[0]}"
+            return generate_username(username_initials)
+        elif 'name' in data:
+            name = data['name'].strip()
+            name_parts = name.split()
+
+            # Take only the first two words for initials
+            if len(name_parts) >= 2:
+                initials = [word[0] for word in name_parts[:2]]
+            else:
```

Comment:
`else` statement would run in case if `name_parts` length is 1,  then can't we directly get first name initial like
`initials = [word[0]]`